cms/articles/tests/test_series_models.py

```diff
@@ -0,0 +1,124 @@
+        article_page = StatisticalArticlePageFactory(parent=self.series)
+        series_response = self.client.get(self.series.url)
+        self.assertEqual(series_response.status_code, 200)
+
+        article_page_response = self.client.get(article_page.url)
+        self.assertEqual(article_page_response.status_code, 200)
+
+        self.assertEqual(series_response.context, article_page_response.context)
+
+    def test_latest_release_external_env(self):
+        """Test latest_release in external env."""
+        StatisticalArticlePageFactory(parent=self.series)
+
+        with override_settings(IS_EXTERNAL_ENV=True):
+            series_response = self.client.get(self.series.url)
+
+        self.assertEqual(series_response.status_code, 200)
+
+
+class ArticleSeriesEvergreenUrlTestCase(WagtailTestUtils, TestCase):
+    def setUp(self):
+        self.article_series_page = ArticleSeriesPageFactory()
+        self.article_with_datasets = StatisticalArticlePageFactory(parent=self.article_series_page)
+        self.article_with_datasets.datasets = StreamValue(
+            DatasetStoryBlock(),
+            stream_data=[
+                (
+                    "manual_link",
+                    {
+                        "title": "Test dataset",
+                        "description": "Test description",
+                        "url": "https://example.com",
+                    },
+                )
+            ],
+        )
+        self.article_with_datasets.save_revision().publish()
+
+    def test_evergreen_route_links_to_evergreen_related_data(self):
+        """Test that the evergreen page links to the evergreen related data page."""
+        response = self.client.get(self.article_series_page.url, follow=True)
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+
+        self.assertContains(
+            response,
+            f'<a href="{self.article_series_page.url}/related-data" class="ons-list__link">'
+            + "View data used in this article</a>",
+            html=True,
+        )
+
+    def test_evergreen_route_related_data_renders_correctly(self):
+        """Check that the expected content is rendered on the evergreen related data page."""
+        response = self.client.get(f"{self.article_series_page.url}/related-data")
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+
+        self.assertContains(response, f"All data related to {self.article_with_datasets.title}")
+        self.assertContains(response, "Test dataset")
+
+    def test_evergreen_route_canonical_url(self):
+        """Test that the canonical URL on the related data page is the evergreen series URL."""
+        response = self.client.get(f"{self.article_series_page.url}/related-data")
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        self.assertContains(
+            response, f'<link rel="canonical" href="{self.article_series_page.get_full_url()}/related-data">', html=True
+        )
+
```

Comment:
Looks like we have some gaps in test, this is testing the evergreen points to everygreen url. Can we also namespace the test to related data. `test_evergreen_route_related_data_canonical_url`

We also need:
- Non evergreen but latest version points to evergreen url
- Non evergreen and not latest point to non evergreen URL